### PR TITLE
Port ApConfigRenew test of test_flow.py to boardfarm

### PR DESCRIPTION
**Pull request by abelog**
## Description
This PR brings port of `ApConfigRenew` test of `test_flows.py` to boardfarm and include it to the general test suite.

## Changes

- Keep MAC of radios in dummy docker and real devices as part of relevant classes. 
- Keep `ucc_socket` which provides methods to communicate with devices over IEEE1905.1 for dummy docker devices and real devices as part of relevant classes.

## Notes

Log of SSID setting was changed comparing to the original `ApConfigRenew` of `test_flows.py` as dummy docker implementation and RAX40 board has different logs about incoming configuration event.
`time.sleep(30)` is required to pass test on RAX40. This should be replaced by wait of unique log in future.

### CODE REVIEW

File: tests/environment.py [hunk 1]
```diff
@@ -458,7 +459,7 @@ def wait_for_log(self, regex: str, start_line: int, timeout: float) -> bool:
         # Multiply timeout by 100, as test sets it in float.
```
Comment (VladyslavTupikin) [inline tests/environment.py:458]: Multiplication was removed but commentary still there.

File: tests/environment.py [hunk 2]
```diff
@@ -471,19 +472,19 @@ def __init__(self, agent: ALEntityPrplWrt, iface_name: str):
...
+        super().__init__(agent, self.mac)
 
-        VirtualAPHostapd(self, mac)
-        VirtualAPHostapd(self, mac)
+        VirtualAPHostapd(self, self.mac)
+        VirtualAPHostapd(self, self.mac)
 
     def wait_for_log(self, regex: str, start_line: int, timeout: float):
         ''' Poll the Radio's logfile until it match regular expression '''
         # Multiply timeout by 100, as test sets it in float.
```
Comment (VladyslavTupikin) [inline tests/environment.py:484]: Multiplication was removed but the commentary still there...

File: tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        controller.ucc_socket.dev_send_1905(agent.mac, 0x000A,
+                                            tlv(0x01, 0x0006, "{" + controller.mac + "}"),
+                                            tlv(0x0F, 0x0001, "{0x00}"),
+                                            tlv(0x10, 0x0001, "{0x00}"))
+
+        time.sleep(30)
+        self.check_log(agent.radios[0],
+                       r"ssid: Multi-AP-24G-1 .*"
+                       r"fronthaul: true backhaul: false",
+                       timeout=60)
```
Comment (VladyslavTupikin) [inline tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py:42]: I think we need to add commentary with an explanation of what 60 does mean, like time in seconds.

**Review by VladyslavTupikin — APPROVED** [review]
Approve, marked changes small and easy to fix.

File: tests/environment.py [hunk 4]
```diff
@@ -279,10 +280,10 @@ def __init__(self, name: str, device: None = None, is_controller: bool = False):
             device_ip = re.search(r'inet (?P<ip>[0-9.]+)',
                                   device_ip_output.decode('utf-8')).group('ip')
 
-        ucc_socket = UCCSocket(device_ip, ucc_port)
-        mac = ucc_socket.dev_get_parameter('ALid')
+        self.ucc_socket = UCCSocket(device_ip, ucc_port)
+        self.mac = self.ucc_socket.dev_get_parameter('ALid')
 
-        super().__init__(mac, ucc_socket, installdir, is_controller)
+        super().__init__(self.mac, self.ucc_socket, installdir, is_controller)
```
Comment (arnout) [inline tests/environment.py]: This makes no sense... `super().__init__(...)` starts with the following code:
```
        self.mac = mac
        self.ucc_socket = ucc_socket
```

So why would you need to set them already in the child class constructor?

It doesn't hurt of course, but I don't like it when something is assigned twice, it makes it hard to understand where it is coming from.

File: tests/environment.py [hunk 5]
```diff
@@ -241,6 +241,7 @@ def _device_wait_for_log(device: None, log_path: str, regex: str,
         matched_line = int(device.match.group('line_number')) + start_line
         return (True, matched_line, match)
     else:
+        err("Can't find {} in log of ".format(regex, device.name))
```
Comment (arnout) [inline tests/environment.py]: This change is unrelated -> separate commit.

File: tests/environment.py [hunk 6]
```diff
@@ -458,7 +459,7 @@ def wait_for_log(self, regex: str, start_line: int, timeout: float) -> bool:
         # Multiply timeout by 100, as test sets it in float.
         return _device_wait_for_log(self.device,
                                     "{}/beerocks_{}.log".format(self.log_folder, program),
-                                    regex, start_line, timeout*100)
+                                    regex, start_line, timeout)
```
Comment (arnout) [inline tests/environment.py:461]: So, now suddenly I'm right and it's actually in seconds rather than centiseconds? :-)

This should be a separate commit.

File: tests/environment.py [hunk 7]
```diff
@@ -471,19 +472,19 @@ def __init__(self, agent: ALEntityPrplWrt, iface_name: str):
         self.iface_name = iface_name
         self.agent = agent
         ip_raw = self.agent.command("ip link list dev {}".format(self.iface_name))
-        mac = re.search(r"link/ether (([0-9a-fA-F]{2}:){5}[0-9a-fA-F]{2})",
-                        ip_raw).group(1)
+        self.mac = re.search(r"link/ether (([0-9a-fA-F]{2}:){5}[0-9a-fA-F]{2})",
+                             ip_raw).group(1)
         self.log_folder = agent.log_folder
-        super().__init__(agent, mac)
+        super().__init__(agent, self.mac)
```
Comment (arnout) [inline tests/environment.py]: Same here, super().__init__() already sets self.mac.

File: tests/environment.py [hunk 8]
```diff
@@ -471,19 +472,19 @@ def __init__(self, agent: ALEntityPrplWrt, iface_name: str):
...
 
-        VirtualAPHostapd(self, mac)
-        VirtualAPHostapd(self, mac)
+        VirtualAPHostapd(self, self.mac)
+        VirtualAPHostapd(self, self.mac)
```
Comment (arnout) [inline tests/environment.py]: Oh, I hadn't noticed before: this is not good! The bssids of the VAPs should be unique. Also, you should use the *actual* BSSIDs of the *actual* VAPs that are created on the device. You will need that to be able to associate stations successfully.

Anyway, fixing that is for another PR.

File: tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    def runTest(self):
+        # Locate agents and controller
+        for dev in self.dev:
+            if dev.controller_entity:
+                controller = dev.controller_entity
+            if dev.agent_entity:
+                agent = dev.agent_entity
+
+        dev.wired_sniffer.start(self.__class__.__name__ + "-" + dev.name)
+        mac_repeater1_upper = agent.mac.upper()
```
Comment (arnout) [inline tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py:26]: Please preserve the comments:
```suggestion
        # Regression test: MAC address should be case insensitive
        mac_repeater1_upper = agent.mac.upper()
```

File: tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+            "DEV_SET_CONFIG,"
+            "bss_info1,{} 8x Multi-AP-24G-1 0x0020 0x0008 maprocks1 0 1,"
+            "bss_info2,{} 8x Multi-AP-24G-2 0x0020 0x0008 maprocks2 1 0"
+            .format(mac_repeater1_upper, agent.mac))
+        controller.ucc_socket.dev_send_1905(agent.mac, 0x000A,
+                                            tlv(0x01, 0x0006, "{" + controller.mac + "}"),
+                                            tlv(0x0F, 0x0001, "{0x00}"),
+                                            tlv(0x10, 0x0001, "{0x00}"))
+
+        time.sleep(30)
```
Comment (arnout) [inline tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py:38]: Autoconfig renew should finish in a few seconds... Does it really take 30 seconds before it finishes?

File: tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py (hunk 3 above)
Comment (arnout) [inline tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py:42]: No, not 30 seconds but *90* seconds? If the default timeout of half a seconds is not enough, we have a problem...

**Review by arnout — APPROVED** [review]
The first commit seems entirely redundant (at least, the parts that are documented in the commit message). I wonder what triggered it?

Except for the timeouts and some little changes, it looks OK.

File: tests/environment.py (hunk 4 above)
Comment (abelog) [reply tests/environment.py]: Yep, i haven't noticed that parent class already assign those fields.  This is determinately redundant change.

File: tests/environment.py (hunk 5 above)
Comment (abelog) [reply tests/environment.py]: Agreed. Removing it. I want to handle timeouts of `_device_wait_for_log` and display more informative message about what message wasn't found in log. Currently, you have to look for exact issue in crash log of boardfarm, which isn't convenient.

File: tests/environment.py (hunk 6 above)
Comment (abelog) [reply tests/environment.py:461]: Yes, you are :smiley:

File: tests/environment.py (hunk 8 above)
Comment (abelog) [reply tests/environment.py]: Separate task created: #1484

File: tests/environment.py (hunk 7 above)
Comment (abelog) [reply tests/environment.py]: Exactly.

File: tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py (hunk 9 above)
Comment (abelog) [reply tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py:26]: Thanks for noticing.

File: tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py (hunk 10 above)
Comment (abelog) [reply tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py:38]: I wasn't doing a lot of experiments to find a minimal value. 30 seconds is enough for RAX40 to pass the test. Do you think is it worth to invest time and find minimal time required for test to pass on RAX40?

File: tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py (hunk 3 above)
Comment (abelog) [reply tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py:42]: There is no great logic behind this 60 sec. timeout, so we might decrease it.

File: tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py (hunk 10 above)
Comment (arnout) [reply tests/boardfarm_plugins/boardfarm_prplmesh/tests/ap_config_renew.py:38]: Well... This is an actual *sleep*, not just a timeout, so we really loose that time. For this one test it's not a big deal, but if we do something like that in every test, our boardfarm tests are going to take *very long*.

And this really should take only a few hundred milliseconds. If it takes longer than that, we probably have a problem somewhere...
